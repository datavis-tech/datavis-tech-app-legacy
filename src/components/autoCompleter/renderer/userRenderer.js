import { Label } from 'semantic-ui-react'

export default ({ title }) => <Label content={title} />
