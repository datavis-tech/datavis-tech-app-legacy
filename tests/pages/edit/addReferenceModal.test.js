/**
 * @jest-environment jsdom
 */

// TODO react-test-renderer does not work with portal so that we need to mock it
// Checkout https://github.com/Semantic-Org/Semantic-UI-React/issues/2454
// and https://github.com/facebook/react/issues/11565 for details
jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children)

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import nodeSelector from '../../utils/nodeSelector'
import { Modal, Form } from 'semantic-ui-react'
import AddReferenceModal from '../../../src/pages/edit/addReferenceModal'

describe('add reference modal', () => {

  let sut
  let props
  let onClose
  let onReferenceSubmit

  beforeEach(() => {
    onClose = jest.fn()
    onReferenceSubmit = jest.fn()

    props = {
      onClose,
      onReferenceSubmit
    }

    sut = shallow(<AddReferenceModal {...props} />)
  })

  it('should not show modal by default', () => {
    expect(sut.find(Modal).props()).toMatchObject({
      open: false,
      onClose,
      size: 'small'
    })
  })

  it('should have cancel button', () => {
    sut.find(nodeSelector('cancel')).simulate('click')
    expect(onClose).toHaveBeenCalled()
  })

  it('should close on close', () => {
    sut.find(Modal).simulate('close')
    expect(onClose).toHaveBeenCalled()
  })

  it('should mark add button as disabled when modal is open', () => {
    sut.setProps({show: true})
    expect(sut.find(nodeSelector('addReference')).props()).toMatchObject({ disabled: true })
  })

  describe('reference addition', () => {

    let filename
    let id

    beforeEach(() => {
      id = String(Math.random())
      filename = String(Math.random())
      sut.setState({ id, filename })
    })

    it('should perform on form submit', () => {
      sut.find(Form).simulate('submit')
      expect(onReferenceSubmit).toHaveBeenCalledWith(filename, id)
    })

    it('should perform when add collaborator button is clicked', () => {
      sut.find(nodeSelector('addReference')).simulate('click')
      expect(onReferenceSubmit).toHaveBeenCalledWith(filename, id)
    })

    it('should mark add button as enabled when both id and filename are filled in', () => {
      expect(sut.find(nodeSelector('addReference')).props()).toMatchObject({
        disabled: false
      })
    })
  })

  it('should render correct layout', () => {

    const userId = String(Math.random())
    const tree = renderer
      .create(<AddReferenceModal userId={userId} show />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
