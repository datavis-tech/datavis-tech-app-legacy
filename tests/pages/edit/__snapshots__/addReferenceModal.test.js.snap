// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add reference modal should render correct layout 1`] = `
<div
  className="ui small modal transition visible active"
  style={
    Object {
      "marginTop": undefined,
    }
  }
>
  <div
    className="content"
  >
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label>
          Filename
        </label>
        <input
          onChange={[Function]}
          placeholder="Type file name (local alias) here."
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label>
          Document id
        </label>
        <div
          className="ui search"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="ui left icon input"
          >
            <input
              autoComplete="off"
              className="prompt"
              disabled={undefined}
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Start typing or paste document ID here."
              tabIndex="0"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
            />
          </div>
          <div
            className="results transition"
          >
            <div
              className="message empty"
            >
              <div
                className="header"
              >
                No results found.
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="actions"
  >
    <button
      className="ui button"
      data-test="cancel"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="remove icon"
      />
      Cancel
    </button>
    <button
      className="ui primary disabled button"
      data-test="addReference"
      disabled={true}
      onClick={[Function]}
      role="button"
      tabIndex={-1}
    >
      <i
        aria-hidden="true"
        className="checkmark icon"
      />
      Add Reference
    </button>
  </div>
</div>
`;
